File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/bridge/HandsetBrand.java
package jv.com.tencent.cain.design_pattern.bridge;

/**
 * @author : jiangyu
 * @date : 2020/7/11
 * @desc : 桥接设计模式-剥离出来的一个分支
 **/
abstract public class HandsetBrand {
    protected HandsetSoft soft;
    protected String brand;

    public HandsetBrand(String brand) {
        this.brand = brand;
    }

    public void setBrand(String brand) {
        this.brand = brand;
    }

    public void setSoft(HandsetSoft soft) {
        this.soft = soft;
    }

    abstract void run();
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/bridge/HandsetBrand.kt
package kt.com.tencent.cain.desigin_pattern.bridge

/**
 * @author : jiangyu
 * @date   : 2020/7/11
 * @desc   : 桥接设计模式-剥离出来的一个分支
 */
abstract class HandsetBrand(var brandName: String, var soft: HandsetSoft? = null) {
    abstract fun run()
}